crypto/src/lib.rs

```diff
@@ -388,6 +364,76 @@ impl PublicKeyInner {
     }
 }
 
+/// `PublicKey` will be lazily deserialized inside WASM.
+/// This is needed for performance reasons, since `PublicKeyInner::from_bytes` is quite slow.
+/// However inside WASM in most cases `PublicKey` is used only for comparisons (==).
+/// See https://github.com/hyperledger/iroha/issues/5038 for details.
+#[cfg(target_family = "wasm")]
+mod lazy {
+    use alloc::{boxed::Box, vec::Vec};
+    use core::{borrow::Borrow, cell::OnceCell};
+
+    use crate::{Algorithm, PublicKeyInner};
+
+    #[derive(Clone, Eq)]
+    pub struct PublicKeyLazy {
+        algorithm: Algorithm,
+        payload: Vec<u8>,
+        inner: OnceCell<Box<PublicKeyInner>>,
```

Comment:
Why is this Box needed?

Reply:
This is to reduce the size of the structure, prior discussion [here](https://github.com/hyperledger/iroha/issues/5038#issuecomment-2334095415)